server/v2/api/grpc/gogoreflection/serverreflection.go

```
@@ -458,26 +458,61 @@ func (s *serverReflectionServer) ServerReflectionInfo(stream rpb.ServerReflectio
+		}
+
 		if _, seen := seenSvc[svc.FullName()]; !seen {
 			svcs = append(svcs, string(svc.FullName()))
 			file := svc.ParentFile()
 			fds = append(fds, protodesc.ToFileDescriptorProto(file))
 		}
+
+		seenSvc[svc.FullName()] = struct{}{}
 	}
+
 	return
```

golangci-lint (1.62.2)

naked return in func `getServices` with 31 lines of code

(nakedret)